scripts/e2e-test.sh

```diff
@@ -3,18 +3,65 @@
 set -eux
 
 SCRIPTDIR=$(dirname "$(realpath "$0")")
-REGISTRY=$1
 TESTS="install basic_volume_io"
+DEVICE=
+REGISTRY=
+TAG=
 
-# TODO: Add proper argument parser
-if [ -z "$REGISTRY" ]; then
-  echo "Missing parameter registry"
+help() {
+  cat <<EOF
+Usage: $(basename $0) [OPTIONS]
+
+Options:
+  --device <path>           Device path to use for storage pools.
+  --registry <host[:port]>  Registry to pull the mayastor images from.
+  --tag <name>              Docker image tag of mayastor images (default "ci")
+
+Examples:
+  $(basename $0) --registry 127.0.0.1:5000 --tag a80ce0c
+EOF
+}
+
+# Parse arguments
+while [ "$#" -gt 0 ]; do
+  case $1 in
+    -d|--device)
+      shift
+      DEVICE=$1
+      shift
+      ;;
+    -r|--registry)
+      shift
+      REGISTRY=$1
+      shift
+      ;;
+    -t|--tag)
+      shift
+      TAG=$1
+      shift
+      ;;
+    *)
+      echo "Unknown option: $1"
+      help
+      exit 1
+      ;;
+  esac
+done
+
+if [ -z "$DEVICE" ]; then
```

Comment:
Readability/maintainability note. I would not rename (i.e. keep the _spelling_) variables throughout the code. I.e. why not use `e2e_pool_device` here? It helps while `grep`-ing through sources as one can see the "path" the value takes throughout system and lowers the cognitive load a little. Please consider not renaming variables between "layers".

Reply:
I'm not sure. variables used by go code are rather long and the current names correspond with the name of shell script arguments which is also good for the reader who tries to understand the code. If we rename them then we should convert e2e variable to uppercase and make sure that empty string is handled equally as undefined variable which I felt is a bit out of scope of this change.

Reply:
TL;DR - I'm resolving the comment as keeping code style isn't really solvable in this one comment :-) Feel free to ignore. I will still provide some thoughts I have around this mainly for the case if you see an error in reasoning and would be willing to provide me with some feedback.

> variables used by go code are rather long; correspond with the name of shell script

I've noticed that envvars are renamed once again inside the go code... back to tag, registry etc go variables. So I would say the whole thing is a problem of env-vars or rather using env-vars instead of cmdline arguments in the _go code_ in the same way as the script does it :+1: (or config). But given the current state if if one searches through env it is nice to see that these random variables are connected to e2e using the prefix that's right (similar to e.g. `MAKE_FLAGS`). Thus I'd say correct solution is to use cmdline args not env vars. Out-of scope of this MR, I agree. And a total nitpick.

>  If we rename them then we should convert e2e variable to uppercase

Well... They should have been uppercase from the start, shouldn't they? But it's kinda random (mostly using uppercase for variables, not constants in shell is common). FYI when I'm in doubt, I tend to fall back to nicely written google style guides: https://google.github.io/styleguide/shellguide.html#s7.2-variable-names https://google.github.io/styleguide/shellguide.html#s7.5-read-only-variables ... And that's how I understand it - constants - which here they are - and should have been uppercase from the start. And variables should be lowercase. But I don't see much value here, unless they're really declared with `readonly <VAR_NAME>` Anyway...

> and make sure that empty string is handled equally as undefined variable which I felt is a bit out of scope of this change

I don't think I follow. Unless you specify `set -u` undefined variable behave as empty ones AFAIK...

Reply:
agree. This whole problem of passing arguments between a caller and the test suites will need to be probably revisited at some point in the future. In fact, currently, it's just the install test suite that requires arguments. But that might change as we add more e2e tests in the future.

re undefined vs empty string: I forgot that there is no difference between that in os.getenv() in go. Please, ignore that comment.

Reply:
Oh right. In golang too (if you dig into it there's`LookupEnv` that can distinguish between unset & empty however building any non-trivial functionality on that distinction is evil :smile:). BTW I really meant in shell, there's no real difference between empty and undefined in scripts (without `set -u`, or unless you again dig deeper with e.g. `env`).